odoo/addons/test_new_api/tests/test_properties.py

```diff
@@ -1346,3 +1346,259 @@ def _get_sql_definition(self, discussion):
+            'value': self.partner.id,
+            'definition_changed': True,
+        }]
+        self.message_2.attributes = {'mypartner': self.partner_2.id}
+        self.message_3.attributes = {'mypartner': False}
+
+        messages = self.env['test_new_api.message'].search(
+            [('attributes.mypartner', 'in', [self.partner.id, self.partner_2.id])])
+        self.assertEqual(messages, self.message_1 | self.message_2)
+
+        messages = self.env['test_new_api.message'].search(
+            [('attributes.mypartner', 'not in', [self.partner.id, self.partner_2.id])])
+        self.assertEqual(messages, self.message_3)
+
+        messages = self.env['test_new_api.message'].search(
+            [('attributes.mypartner', 'ilike', self.partner.display_name)])
+        self.assertFalse(messages, "The ilike on relational properties is not supported")
+
+    @mute_logger('odoo.fields')
+    def test_properties_field_search_boolean(self):
+        # search on boolean
+        self.message_1.attributes = [{
+            'name': 'myboolean',
+            'type': 'boolean',
+            'value': True,
+            'definition_changed': True,
+        }]
+        self.message_2.attributes = {'myboolean': False}
+        messages = self.env['test_new_api.message'].search([('attributes.myboolean', '=', True)])
+        self.assertEqual(messages, self.message_1)
+        messages = self.env['test_new_api.message'].search([('attributes.myboolean', '=', False)])
+        # message 2 has a falsy boolean properties
+        # message 3 doesn't have the properties (key in dict doesn't exist)
+        self.assertEqual(messages, self.message_2 | self.message_3)
+
+    @mute_logger('odoo.fields')
+    def test_properties_field_search_tags(self):
+        self.messages.discussion = self.discussion_1
+        self.message_1.attributes = [{
+            'name': 'mytags',
+            'type': 'tags',
+            'value': ['a', 'b'],
+            'tags': [['a', 'A', 1], ['b', 'B', 2], ['aa', 'AA', 3]],
+            'definition_changed': True,
+        }]
+        self.message_2.attributes = {'mytags': ['b']}
+        self.message_3.attributes = {'mytags': ['aa']}
+
+        messages = self.env['test_new_api.message'].search([('attributes.mytags', 'in', 'a')])
+        self.assertEqual(messages, self.message_1)
+        # the search is done on the JSONified value (operator "->>")
+        messages = self.env['test_new_api.message'].search([('attributes.mytags', 'ilike', 'a')])
+        self.assertEqual(messages, self.message_1 | self.message_3)
+        messages = self.env['test_new_api.message'].search([('attributes.mytags', 'not ilike', 'a')])
+        self.assertEqual(messages, self.message_2)
+        messages = self.env['test_new_api.message'].search([('attributes.mytags', 'in', 'b')])
+        self.assertEqual(messages, self.message_1 | self.message_2)
+        messages = self.env['test_new_api.message'].search([('attributes.mytags', 'in', 'aa')])
```

Comment:
actually for the tags it work because it's a list of string and not a list of integer like the many2many :disguised_face: (CF the difference between the operator on list of string VS list of integer)

ok, I add tests even if it's not used by the web client